### PR TITLE
Fix FrameworkActions not being handled in some scenarios

### PR DESCRIPTION
**Pull request by nyquillerium**
Fixes https://github.com/ppy/osu-framework/issues/2671

Also makes `FrameworkActionContainer` internal.

### CODE REVIEW

**Comment by peppy** [comment]
Need more details on how/why this happened. Alone this PR doesn't really give enouhg explanation.

**Comment by nyquillerium** [comment]
The game relied on being prioritized by the input queue at the `FrameworkActionContainer` previously, and this broke once the Game was no longer a direct child of `FrameworkActionContainer`. Additionally, `OverlayContainer` removes all previous Drawables in the input queue that are not of `IHandleGlobalInput`, so this would cause the game to be removed from the Draw Hierarchy every time an overlay is open.

I've gone ahead and made `Game` use `IHandleGlobalInput` in addition to this change.

File: osu.Framework/Input/FrameworkActionContainer.cs [hunk 1]
```diff
@@ -19,9 +19,19 @@ public class FrameworkActionContainer : KeyBindingContainer<FrameworkAction>
             new KeyBinding(new[] { InputKey.Alt, InputKey.Enter }, FrameworkAction.ToggleFullscreen),
         };
 
+        private readonly Game game;
+
+        public FrameworkActionContainer(Game game = null)
```
Comment (peppy) [inline osu.Framework/Input/FrameworkActionContainer.cs]: isn't this going to break if games is null

Comment (nyquillerium) [reply osu.Framework/Input/FrameworkActionContainer.cs]: It isn't breaking in the other scenario where this is used in the mouse test, but I'll make `KeyBindingInputQueue` handle null for safety

Comment (peppy) [reply osu.Framework/Input/FrameworkActionContainer.cs]: No.

Just don't make it nullable.

Comment (nyquillerium) [reply osu.Framework/Input/FrameworkActionContainer.cs]: so throw if game is null?

Comment (ShineLoliMC) [reply osu.Framework/Input/FrameworkActionContainer.cs]: This page is so funny

**Pull request merged**